---
title: "CCA Y RDA"
author:
- name: Stephanie Hereira
  affiliation: Universidad Aut√≥noma de Tlaxcala
  email: sehereirap@uatx.mx
date: "`r format(Sys.time(), '%d - %m - %Y')`"
output:
  html_document:
    theme: flatly
    highlight: kate
    toc: yes
    toc_depth: 2
    toc_float: yes
  pdf_document:
    toc: yes
    toc_depth: '2'
---

## Figurs 6 and 7: cca
- Loading packages and data
```{r, warning=FALSE, message=FALSE}
knitr::opts_chunk$set(echo = T, results = "hide")
library(reshape2)
library(tidyverse)
library(picante)
library(Rmisc)
library(readxl)
library(vegan)
library(qiime2R)
source("../Code/functions_betadiv.R")

#Environmental data
metadata_secas<- read_excel("../Data/Metadatos.xlsx", sheet = "secas-marzo")
fq_secas<- read_excel("../Data/fisicoq.xlsx", sheet = "seca")
fq_secas2<- read.csv("../Data/fisicoq-la.csv")
meta_fq_secas<- metadata_secas %>% full_join(fq_secas) 
meta_fq_secas_all<- metadata_secas %>% full_join(fq_secas) %>% full_join(fq_secas2, by = "SampleID")
env<-meta_fq_secas_all

metadata<-read_excel("../Data/Metadatos.xlsx") %>% mutate_if(is.numeric, as.factor)


# Vegetation data

veg<- read_tsv("../Data/vegeta.tsv") %>% as.data.frame( ) %>% 
   select_all(~str_replace(., "prop_", ""))  %>% 
   select_all(~str_replace(., "mean_", ""))  %>% 
   select_all(~str_replace(., "coverage_", "cov_")) %>% mutate(SampleID=case_when(
  SampleID=="P6S1T1" ~"P6S2T1",
  SampleID=="P6S1T2" ~"P6S2T2",
  SampleID=="P6S1T3" ~"P6S2T3",
  SampleID=="P6S2T1" ~"P6S1T1",
  SampleID=="P6S2T2" ~"P6S1T2",
  SampleID=="P6S2T3" ~"P6S1T3",
  TRUE~as.character(SampleID))) %>% arrange(SampleID)

# Metagenomic data

table_single<-read.delim("../Data/table_micop_single.txt") 
table_paired<-read.delim("../Data/table_micop_paired.txt") 
table_kraken<-read.delim("../Data/table_fungi_again.txt", 
                   skip = 1,
                   row.names = 1, check.names = F) %>% dplyr::select_if(
                     is.numeric)%>% t() %>% as.data.frame() %>% rownames_to_column(var = "id_sequence") %>% separate(
                         ., "id_sequence", c(
                           "kraken", "fungi", 
                           "id_metagenome", "report", "bracken"), 
                         sep = "_") %>% dplyr::select(
                           -kraken, -fungi, -report, -bracken) %>% full_join(
                           metadata) %>% dplyr::select(-id_sequence:-Transecto, -id_metagenome, -Sites, -id_new, -Names,  -id_fisicoq) %>% column_to_rownames(var = "SampleID") %>% t() %>% as.data.frame() %>% mutate_all(as.numeric)

table_qiime2<-data.frame(read_qza("../Data/clustered_table_filter.qza")$data, 
                   check.names = F) %>% t() %>% as.data.frame(
                   ) %>% rownames_to_column(
                     var = "SampleID") %>% separate(
                       SampleID, c(
                         "id_metagenome", "R", "unmap", "Paired"), 
                       sep = "_")%>% inner_join(
                         metadata) %>% dplyr::select(
                           -id_metagenome:-Paired, 
                           -id_sequence:-id_fisicoq, -Sites, 
                           -Names) %>% column_to_rownames(
                             var="SampleID") %>% t(
                             ) %>% as.data.frame() %>% mutate_all(as.numeric)



table_qiime2_t<-data.frame(t(table_qiime2), check.names = F)
table_single_t<-data.frame(t(table_single), check.names = F)
table_paired_t<-data.frame(t(table_paired), check.names = F)
table_kraken_t<-data.frame(t(table_kraken), check.names = F)
```

- Environmental analysis
```{r, warning=FALSE, message=FALSE}

#select variables that did not show collinearity and order data
set.seed(124)
vars<- c("MO", "Ca", "LIMO", "ARCILLA", "WHC", "Fe", "K", "Mg",
         "pH", "Cu", "P", "N")
#qiime2
mm1<-table_qiime2_t %>% rownames_to_column(var = "SampleID") %>% inner_join(
  env) %>% column_to_rownames(var = "SampleID")
env1=mm1[,vars]
#env1=mm1[,c(3149:3159, 3162,  3165,3167)]
spp1=mm1[,1:3138]

#single
mm2<-table_single_t %>% rownames_to_column(var = "SampleID") %>% inner_join(
  env) %>% column_to_rownames(var = "SampleID")
env2=mm2[,vars]
spp2=mm2[,1:210]

#paired
mm3<-table_paired_t %>% rownames_to_column(var = "SampleID") %>% inner_join(
  env) %>% column_to_rownames(var = "SampleID")
env3=mm3[,vars]
spp3=mm3[,1:234]

#kraken
mm4<-table_kraken_t %>% rownames_to_column(var = "SampleID") %>% inner_join(
  env) %>% column_to_rownames(var = "SampleID")
env4=mm4[,vars]
spp4=mm4[,1:72]


# Hellinger transformation
spp_hell1=decostand(spp1, "hell")
spp_hell2=decostand(spp2, "hell")
spp_hell3=decostand(spp3, "hell")
spp_hell4=decostand(spp4, "hell")



# Transforming environmental data
envscho<-c("MO", "Ca", "Silt", "WHC", "Fe", "K", "Mg", "pH", "Cu", "Clay", "P",  "N")
envst1=data.frame(scale(env1, scale=T, center=F)) %>% dplyr::rename(Silt=LIMO, #Humidity=moisture, 
                                                                  Clay=ARCILLA) %>% dplyr::select(all_of(envscho))
envst2=data.frame(scale(env2, scale=T, center=F))%>% dplyr::rename(Silt=LIMO, #Humidity=moisture, 
                                                                   Clay=ARCILLA) %>% dplyr::select(all_of(envscho))
envst3=data.frame(scale(env3, scale=T, center=F))%>% dplyr::rename(Silt=LIMO, #Humidity=moisture, 
                                                                   Clay=ARCILLA) %>% dplyr::select(all_of(envscho))
envst4=data.frame(scale(env4, scale=T, center=F))%>% dplyr::rename(Silt=LIMO, #Humidity=moisture, 
                                                                   Clay=ARCILLA) %>% dplyr::select(all_of(envscho))

# Forward selection procedure


#selecting using cca (no distance)
set.seed(126)
vares_cca1 <- cca(spp_hell1 ~., data=envst1)
envs1<-envfit(vares_cca1 ~ ., data=envst1)$vectors[["pvals"]]%>% as.data.frame() %>% dplyr::select("QIIME2"=".")

vares_cca2 <- cca(spp_hell2 ~., data=envst2)
envs2<-envfit(vares_cca2 ~ ., data=envst2)$vectors[["pvals"]]%>% as.data.frame() %>% dplyr::select("Single Micop"=".")

vares_cca3 <- cca(spp_hell3 ~.,data=envst3)
envs3<-envfit(vares_cca3 ~ ., data=envst3)$vectors[["pvals"]]%>% as.data.frame() %>% dplyr::select("Paired Micop"=".")

vares_cca4 <- cca(spp_hell4 ~.,data=envst4)
envs4<-envfit(vares_cca4 ~ ., data=envst4)$vectors[["pvals"]]%>% as.data.frame() %>% dplyr::select("Kraken2"=".")

envs<- cbind(envs1, envs2, envs3, envs4)
rownames(envs)
envs

#accounting for abundance
cap.env1=capscale(spp_hell1~.,envst1,distance = "bray")
mod0.env1=capscale(spp_hell1~1,envst1,distance = "bray")
step.env1=ordistep(mod0.env1,scope = formula(cap.env1))
cap1<-step.env1$anova %>% mutate(method="QIIME2") #no significants
meta1<- env1 %>% rownames_to_column(var = "SampleID") %>%  inner_join(metadata)

cap.env2=capscale(spp_hell2~.,envst2,distance = "bray")
mod0.env2=capscale(spp_hell2~1,envst2,distance = "bray")
step.env2=ordistep(mod0.env2,scope = formula(cap.env2))
cap2<-step.env2$anova %>% mutate(method="Single Micop") 
meta2<- env2 %>% rownames_to_column(var = "SampleID") %>%  inner_join(metadata)


cap.env3=capscale(spp_hell3~.,envst3,distance = "bray")
mod0.env3=capscale(spp_hell3~1,envst3,distance = "bray")
step.env3=ordistep(mod0.env3,scope = formula(cap.env3))
cap3<-step.env3$anova %>% mutate(method="Paired Micop") 
meta3<- env3 %>% rownames_to_column(var = "SampleID") %>%  inner_join(metadata)

cap.env4=capscale(spp_hell4~.,envst4,distance = "bray")
mod0.env4=capscale(spp_hell4~1,envst4,distance = "bray")
step.env4=ordistep(mod0.env4,scope = formula(cap.env4))
cap4<-step.env4$anova %>% mutate(method="Kraken2") 
meta4<- env4 %>% rownames_to_column(var = "SampleID") %>%  inner_join(metadata)

#horn
cap.env1_c=capscale(spp_hell1~.,envst1,distance = "horn")
mod0.env1_c=capscale(spp_hell1~1,envst1,distance = "horn")
step.env1_c=ordistep(mod0.env1_c,scope = formula(cap.env1_c))
cap1_c<-step.env1_c$anova %>% mutate(method="QIIME2")

cap.env2_c=capscale(spp_hell2~.,envst2,distance = "horn")
mod0.env2_c=capscale(spp_hell2~1,envst2,distance = "horn")
step.env2_c=ordistep(mod0.env2_c,scope = formula(cap.env2_c))
cap2_c<-step.env2_c$anova %>% mutate(method="Single Micop") 

cap.env3_c=capscale(spp_hell3~.,envst3,distance = "horn")
mod0.env3_c=capscale(spp_hell3~1,envst3,distance = "horn")
step.env3_c=ordistep(mod0.env3_c,scope = formula(cap.env3_c))
#cap3_c<-step.env3_c$anova %>% mutate(method="Paired Micop") 

cap.env4_c=capscale(spp_hell4~.,envst4,distance = "horn")
mod0.env4_c=capscale(spp_hell4~1,envst4,distance = "horn")
step.env4_c=ordistep(mod0.env4_c,scope = formula(cap.env4_c))
cap4_c<-step.env4_c$anova %>% mutate(method="Kraken2")  #no significant

```

- Data for plotting and plots


```{r, message=FALSE, warning=FALSE}
library(ggvegan)

map<- read.csv("../Data/coord.csv") %>% mutate_at(
  c(1,2,3,7), as.factor) %>% mutate(SampleID= paste0("P",pol, "S", Sitio,"T", Transecto ))


#a1
fdat <- ggplot2::fortify(vares_cca1)
map1<- map[match(rownames(spp_hell1), map$SampleID),]
map1_type<- map1 %>% dplyr::select(Site, Type) %>% group_by(Site, Type) %>% count()
  
y<-ggordiplots::gg_ordiplot(vares_cca1, groups = map1$Site, ellipse =  F, 
                            spiders = T, hull = F, plot = F )
z <- y$plot
a1<-z+theme_linedraw()+
  geom_vline(xintercept = 0, linetype = 2) +   #lines-cross
  geom_hline(yintercept = 0, linetype = 2) +
  theme_linedraw()+
  scale_color_viridis_d(option ="turbo" )+#color of points +
    ggrepel::geom_text_repel(data = fdat %>% filter(
  score=="biplot") %>%  
      mutate(CCA1=CCA1*1.9, CCA2=CCA2*1.9) ,
    aes(x=CCA1, y=CCA2, label= label),
    color="red", segment.color = NA, size=4.5)+
  theme_linedraw() +
  theme(axis.text = element_text(colour = "black", size = 5),
        axis.title = element_text(colour = "black", size = 10),
        legend.text = element_text(size = 10),
        legend.title = element_text(size = 10), 
        legend.position = "right", 
        legend.box = "vertical",
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank())+
  geom_vline(xintercept = 0, linetype = 2) +   #lines-cross
  geom_hline(yintercept = 0, linetype = 2) +
  geom_label(
    data = y$df_mean.ord,
    aes(x = x, y = y, label=Group), 
    label.padding = unit(0.15, "lines"),label.size = 0.4,
  )+guides(
    color=guide_legend(title="Sites"))+ylab("CCA2")+xlab("CCA1")+
  theme(axis.text = element_text(size = 12))
  

#a2
fdat <- ggplot2::fortify(vares_cca2)
map2<- map[match(rownames(spp_hell2), map$SampleID),]
map2_type<- map2 %>% dplyr::select(Site, Type) %>% group_by(Site, Type) %>% count()

y<-ggordiplots::gg_ordiplot(vares_cca2, groups = map2$Site, ellipse =  F, 
                            spiders = T, hull = F, plot = F )
z <- y$plot
a2<-z+theme_linedraw()+
  geom_vline(xintercept = 0, linetype = 2) +   #lines-cross
  geom_hline(yintercept = 0, linetype = 2) +
  theme_linedraw()+
  scale_color_viridis_d(option ="turbo" )+#color of points +
  ggrepel::geom_text_repel(data = fdat %>% filter(
    score=="biplot") %>%  
      mutate(CCA1=CCA1*0.6, CCA2=CCA2*0.6) ,
    aes(x=CCA1, y=CCA2, label= label),
    color="red", segment.color = NA, size=4.5)+
  theme_linedraw() +
  theme(axis.text = element_text(colour = "black", size = 5),
        axis.title = element_text(colour = "black", size = 10),
        legend.text = element_text(size = 10),
        legend.title = element_text(size = 10), 
        legend.position = "right", 
        legend.box = "vertical",
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank())+
  geom_vline(xintercept = 0, linetype = 2) +   #lines-cross
  geom_hline(yintercept = 0, linetype = 2) +
  geom_label(
    data = y$df_mean.ord,
    aes(x = x, y = y, label=Group), 
    label.padding = unit(0.15, "lines"),label.size = 0.4,
  )+guides(
    color=guide_legend(title="Sites"))+ylab("CCA2")+xlab("CCA1")+
  theme(axis.text = element_text(size = 12))
#a3
fdat <- ggplot2::fortify(vares_cca3)
map3<- map[match(rownames(spp_hell3), map$SampleID),]
map3_type<- map2 %>% dplyr::select(Site, Type) %>% group_by(Site, Type) %>% count()

y<-ggordiplots::gg_ordiplot(vares_cca3, groups = map3$Site, ellipse =  F, 
                            spiders = T, hull = F, plot = F )
z <- y$plot
a3<-z+theme_linedraw()+
  geom_vline(xintercept = 0, linetype = 2) +   #lines-cross
  geom_hline(yintercept = 0, linetype = 2) +
  theme_linedraw()+
  scale_color_viridis_d(option ="turbo" )+#color of points +
  ggrepel::geom_text_repel(data = fdat %>% filter(
    score=="biplot") %>%  
      mutate(CCA1=CCA1*0.9, CCA2=CCA2*0.9) ,
    aes(x=CCA1, y=CCA2, label= label),
    color="red", segment.color = NA, size=4.5)+
  theme_linedraw() +
  theme(axis.text = element_text(colour = "black", size = 5),
        axis.title = element_text(colour = "black", size = 10),
        legend.text = element_text(size = 10),
        legend.title = element_text(size = 10), 
        legend.position = "right", 
        legend.box = "vertical",
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank())+
  geom_vline(xintercept = 0, linetype = 2) +   #lines-cross
  geom_hline(yintercept = 0, linetype = 2) +
  geom_label(
    data = y$df_mean.ord,
    aes(x = x, y = y, label=Group), 
    label.padding = unit(0.15, "lines"),label.size = 0.4,
  )+guides(
    color=guide_legend(title="Sites"))+ylab("CCA2")+xlab("CCA1")+
  theme(axis.text = element_text(size = 12))


#a4
fdat <- ggplot2::fortify(vares_cca4)
map4<- map[match(rownames(spp_hell4), map$SampleID),]
map4_type<- map4 %>% dplyr::select(Site, Type) %>% group_by(Site, Type) %>% dplyr::count()

y<-ggordiplots::gg_ordiplot(vares_cca4, groups = map4$Site, ellipse =  F, 
                            spiders = T, hull = F, plot = F )
z <- y$plot
a4<-z+theme_linedraw()+
  geom_vline(xintercept = 0, linetype = 2) +   #lines-cross
  geom_hline(yintercept = 0, linetype = 2) +
  theme_linedraw()+
  scale_color_viridis_d(option ="turbo" )+#color of points +
  ggrepel::geom_text_repel(data = fdat %>% filter(
    score=="biplot") %>%  
      mutate(CCA1=CCA1*0.1, CCA2=CCA2*0.1) ,
    aes(x=CCA1, y=CCA2, label= label),
    color="red", segment.color = NA, size =4.5)+
  theme_linedraw() +
  theme(axis.text = element_text(colour = "black", size = 5),
        axis.title = element_text(colour = "black", size = 10),
        legend.text = element_text(size = 10),
        legend.title = element_text(size = 10), 
        legend.position = "right", 
        legend.box = "vertical",
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank())+
  geom_vline(xintercept = 0, linetype = 2) +   #lines-cross
  geom_hline(yintercept = 0, linetype = 2) +
  geom_label(
    data = y$df_mean.ord,
    aes(x = x, y = y, label=Group), 
    label.padding = unit(0.15, "lines"),label.size = 0.4,
  )+guides(
    color=guide_legend(title="Sites"))+ylab("CCA2")+xlab("CCA1")+
  theme(axis.text = element_text(size = 12))




library(cowplot)
as<-plot_grid(a1+theme(aspect.ratio =6/10, legend.position = "none"),
                   a2+theme(aspect.ratio =6/10, legend.position = "none"),
                   a3+theme(aspect.ratio =6/10, legend.position = "none"),
                   a4+theme(aspect.ratio =6/10, legend.position = "none"),
              nrow = 2, ncol = 2,  align = "v", rel_widths = c(1,1,1,1),
              label_size = 15,
                  labels = c("A) QIIME2", "B) SINGLE MICOP",
                             "C) PAIRED MICOP", "D) KRAKEN2"),hjust = 0)
as
ggsave("cca_ggplot.jpg",width = 12, height =8.5, dpi = 300, plot = as, device = "jpg")

```



```{r}
taxonomy_qiime2<- data.frame(read_qza("../Data/taxonomy_blast_dfc_0.98.qza")$data, check.names = F) %>% dplyr::select(Feature.ID,Taxon)
taxonomy_single_micop<- read.delim("../Data/table_micop_single.txt") %>% rownames_to_column(var = "Feature.ID") %>% dplyr::select(Feature.ID) %>% mutate(Taxon=Feature.ID)
taxonomy_paired_micop<- read.delim("../Data/table_micop_paired.txt") %>% rownames_to_column(var = "Feature.ID") %>% dplyr::select(Feature.ID) %>% mutate(Taxon=Feature.ID)
taxonomy_fungi<- read.delim("../Data/table_fungi_again.txt",
                            skip = 1, row.names = 1, check.names = F) %>% select_if(is.character) %>% rownames_to_column(
                              var = "#OTU ID") %>% dplyr::select(Feature.ID = "#OTU ID", Taxon= taxonomy)

map<- read.csv("../Data/coord.csv") %>% mutate_at(
  c(1,2,3,7), as.factor) %>% mutate(SampleID= paste0("P",pol, "S", Sitio,"T", Transecto ))

```

- Plot ccas envs
```{r, message=FALSE, warning=FALSE}

#png("cap_envs.png",width=6,height=4,units="in",res=1200)

library(viridis)
pal<- viridis(12, option = "H") 
pal2<- c( "#30123BFF","#30123BFF","#30123BFF","#4454C4FF","#4454C4FF","#4454C4FF",
                     "#4490FEFF","#4490FEFF","#4490FEFF" ,"#1FC8DEFF","#1FC8DEFF","#1FC8DEFF",
                     "#29EFA2FF","#29EFA2FF","#29EFA2FF", "#7DFF56FF","#7DFF56FF","#7DFF56FF",
                     "#C1F334FF","#C1F334FF","#C1F334FF", "#F1CA3AFF", "#F1CA3AFF", "#F1CA3AFF",
                     "#FE922AFF","#FE922AFF","#FE922AFF" , "#EA4F0DFF", "#EA4F0DFF", "#EA4F0DFF",
                     "#BE2102FF","#BE2102FF","#BE2102FF", "#7A0403FF","#7A0403FF","#7A0403FF")
                     #plots
par(mfrow=c(2,2),mar=c(2, 2, 1,1))

ordiplot(vares_cca1,type="n", main="QIIME2", cex.main=1)
text(cap.env1, dis="cn", scaling="sites", cex=0.6, col="red")
#points(cap.env1,display="sites",cex=0.5,pc=19,  col = pal2)
ordiellipse(vares_cca1, groups =meta1$Sites, col =pal, display = "sites",
            label=T, draw = "polygon", alpha = 50, label.cex=0.5, cex=0.7)
#text(x = vars_choosing1$CAP1,y = vars_choosing1$CAP2, 
#    cex=0.8, col="blue", labels=vars_choosing1$tax, font=3)
#text(cap.env1, dis="species", scaling="sites", cex=0.5, col="black")


ordiplot(vares_cca1,type="n", main="SINGLE MICOP", cex.main=1)
text(cap.env2, dis="cn", scaling="sites", cex=0.6, col="red")
#points(cap.env2,display="sites",cex=0.5,pc=19,  col = pal2)
ordiellipse(cap.env2, groups =meta2$Sites, col =pal, display = "sites",
            label=T, draw = "polygon", alpha = 50, label.cex=0.5, cex=0.7)
#text(x = vars_choosing2$CAP1,y = vars_choosing2$CAP2, 
#    cex=0.8, col="blue", labels=vars_choosing2$tax, font=3)

ordiplot(vares_cca1,type="n", main="PAIRED MICOP", cex.main=1)
text(cap.env3, dis="cn", scaling="sites", cex=0.6, col="red")
#points(cap.env3,display="sites",cex=0.5,pc=19,  col = pal2)
ordiellipse(cap.env3, groups =meta3$Sites, col =pal, display = "sites",
            label=T, draw = "polygon", alpha = 50, label.cex=0.5, cex=0.7)

#text(x = vars_choosing3$CAP1,y = vars_choosing3$CAP2, 
#    cex=0.8, col="blue", labels=vars_choosing3$tax, font=3)

ordiplot(cap.env4,type="n", main="KRAKEN2", cex.main=1)
text(cap.env4, dis="cn", scaling="sites", cex=0.6, col="red")
#points(cap.env4,display="sites",cex=0.5,pc=19,  col = pal2)
ordiellipse(cap.env4, groups =meta4$Sites, col =pal, display = "sites",
            label=T, draw = "polygon", alpha = 50, label.cex=0.5, cex=0.7)
#text(x = vars_choosing4$CAP1,y = vars_choosing4$CAP2, 
#    cex=0.8, col="blue", labels=vars_choosing4$tax, font=3)



#dev.off()


ordiplot(cap.env1_c,type="n", main="QIIME2", cex.main=1)
text(cap.env1_c, dis="cn", scaling="sites", cex=0.6, col="red")
#points(cap.env1_c,display="sites",cex=0.5,pc=19,  col = pal2)
ordiellipse(cap.env1_c, groups =meta1$Sites, col =pal, display = "sites",
            label=T, draw = "polygon", alpha = 50, label.cex=0.5, cex=0.7)
#text(x = vars_choosing1$CAP1,y = vars_choosing1$CAP2, 
#    cex=0.8, col="blue", labels=vars_choosing1$tax, font=3)
#text(cap.env1_c, dis="species", scaling="sites", cex=0.5, col="black")


ordiplot(cap.env2_c,type="n",main="SINGLE MICOP", cex.main=1)
text(cap.env2_c, dis="cn", scaling="sites", cex=0.6, col="red")
#points(cap.env2_c,display="sites",cex=0.5,pc=19,  col = pal2)
ordiellipse(cap.env2_c, groups =meta2$Sites, col =pal, display = "sites",
            label=T, draw = "polygon", alpha = 50, label.cex=0.5, cex=0.7)
#text(x = vars_choosing2$CAP1,y = vars_choosing2$CAP2, 
#    cex=0.8, col="blue", labels=vars_choosing2$tax, font=3)

ordiplot(cap.env3_c,type="n",main="PAIRED MICOP", cex.main=1)
text(cap.env3_c, dis="cn", scaling="sites", cex=0.6, col="red")
#points(cap.env3_c,display="sites",cex=0.5,pc=19,  col = pal2)
ordiellipse(cap.env3_c, groups =meta3$Sites, col =pal, display = "sites",
            label=T, draw = "polygon", alpha = 50, label.cex=0.5, cex=0.7)

#text(x = vars_choosing3$CAP1,y = vars_choosing3$CAP2, 
#    cex=0.8, col="blue", labels=vars_choosing3$tax, font=3)

ordiplot(cap.env4_c,type="n",main="KRAKEN2", cex.main=1)
text(cap.env4_c, dis="cn", scaling="sites", cex=0.6, col="red")
#points(cap.env4_c,display="sites",cex=0.5,pc=19,  col = pal2)
ordiellipse(cap.env4_c, groups =meta4$Sites, col =pal, display = "sites",
            label=T, draw = "polygon", alpha = 50, label.cex=0.5, cex=0.7)
#text(x = vars_choosing4$CAP1,y = vars_choosing4$CAP2, 
#    cex=0.8, col="blue", labels=vars_choosing4$tax, font=3)



```



- Vegetation analysis

```{r, warning=FALSE, message=FALSE}
#qiime2
mm1= veg %>% inner_join(table_qiime2_t %>%rownames_to_column(var = "SampleID"))
veg1=mm1[1:15] %>% column_to_rownames(var = "SampleID")
spp1=mm1[c(1,16:3153)]%>% column_to_rownames(var = "SampleID")

#single
mm2=veg%>% inner_join(table_single_t %>%rownames_to_column(var = "SampleID") )
veg2=mm2[1:15] %>% column_to_rownames(var = "SampleID")
spp2=mm2[c(1,16:225)]%>% column_to_rownames(var = "SampleID")

#paired
mm3=veg %>% inner_join(table_paired_t %>%rownames_to_column(var = "SampleID") )
veg3=mm3[1:15] %>% column_to_rownames(var = "SampleID")
spp3=mm3[c(1,16:249)]%>% column_to_rownames(var = "SampleID")

#kraken
mm4=veg %>% inner_join(table_kraken_t %>%rownames_to_column(var = "SampleID") )
veg4=mm2[1:15] %>% column_to_rownames(var = "SampleID")
spp4=mm4[c(1,16:87)]%>% column_to_rownames(var = "SampleID")


# Hellinger transformation

spp_hell1=decostand(spp1, "hell") 
spp_hell2=decostand(spp2, "hell")
spp_hell3=decostand(spp3, "hell")
spp_hell4=decostand(spp4, "hell")

# Compositional tranformation
#spp_hell1=decostand(otu.single(spp1), "clr", pseudocount=0.5) 
#spp_hell2=decostand(otu.single(spp2),  "clr", pseudocount=0.5)
#spp_hell3=decostand(otu.single(spp3),  "clr", pseudocount=0.5)
#spp_hell4=decostand(otu.single(spp4),  "clr", pseudocount=0.5)

# Transforming environmental data
vegt1=data.frame(scale(veg1, scale=T, center=F)) 
vegt2=data.frame(scale(veg2, scale=T, center=F))
vegt3=data.frame(scale(veg3, scale=T, center=F))
vegt4=data.frame(scale(veg4, scale=T, center=F))

# Forward selection procedure
#accounting for abundance
cap.veg1=capscale(spp_hell1~.,vegt1,distance = "bray")
mod0.veg1=capscale(spp_hell1~1,vegt1,distance = "bray")
step.veg1=ordistep(mod0.veg1,scope = formula(cap.veg1))
cap1v<-step.veg1$anova %>% mutate(method="QIIME2")
meta1<- veg1 %>% rownames_to_column(var = "SampleID") %>%  inner_join(metadata)

cap.veg2=capscale(spp_hell2~.,vegt2,distance = "bray")
mod0.veg2=capscale(spp_hell2~1,vegt2,distance = "bray")
step.veg2=ordistep(mod0.veg2,scope = formula(cap.veg2))
cap2v<-step.veg2$anova %>% mutate(method="Single Micop") 
meta2<- veg2 %>% rownames_to_column(var = "SampleID") %>%  inner_join(metadata)

cap.veg3=capscale(spp_hell3~.,vegt3,distance = "bray")
mod0.veg3=capscale(spp_hell3~1,vegt3,distance = "bray")
step.veg3=ordistep(mod0.veg3,scope = formula(cap.veg3))
cap3v<-step.veg3$anova %>% mutate(method="Paired Micop") 
meta3<- veg3 %>% rownames_to_column(var = "SampleID") %>%  inner_join(metadata)

cap.veg4=capscale(spp_hell4~.,vegt4,distance = "bray")
mod0.veg4=capscale(spp_hell4~1,vegt4,distance = "bray")
step.veg4=ordistep(mod0.veg4,scope = formula(cap.veg4))
cap4v<-step.veg4$anova %>% mutate(method="Kraken2") 
meta4<- veg4 %>% rownames_to_column(var = "SampleID") %>%  inner_join(metadata)

#horn
cap.veg1_c=capscale(spp_hell1~.,vegt1,distance = "horn")
mod0.veg1_c=capscale(spp_hell1~1,vegt1,distance = "horn")
step.veg1_c=ordistep(mod0.veg1_c,scope = formula(cap.veg1_c))
cap1_vc<-step.veg1_c$anova %>% mutate(method="QIIME2")

cap.veg2_c=capscale(spp_hell2~.,vegt2,distance = "horn")
mod0.veg2_c=capscale(spp_hell2~1,vegt2,distance = "horn")
step.veg2_c=ordistep(mod0.veg2_c,scope = formula(cap.veg2_c))
cap2_vc<-step.veg2_c$anova %>% mutate(method="Single Micop") 

cap.veg3_c=capscale(spp_hell3~.,vegt3,distance = "horn")
mod0.veg3_c=capscale(spp_hell3~1,vegt3,distance = "horn")
step.veg3_c=ordistep(mod0.veg3_c,scope = formula(cap.veg3_c))
cap3_vc<-step.veg3_c$anova %>% mutate(method="Paired Micop") 

cap.veg4_c=capscale(spp_hell4~.,vegt4,distance = "horn")
mod0.veg4_c=capscale(spp_hell4~1,vegt4,distance = "horn")
step.veg4_c=ordistep(mod0.veg4_c,scope = formula(cap.veg4))
cap4_vc<-step.veg4_c$anova %>% mutate(method="Kraken2") 

#selecting using cca (no distance)
vares_cca1_veg <- cca(spp_hell1 ~., data=vegt1)
vegs1<-envfit(vares_cca1_veg ~., data=vegt1)$vectors[["pvals"]]%>% as.data.frame() %>% dplyr::select("QIIME2"=".")

vares_cca2_veg <- cca(spp_hell2 ~., data=vegt2)
vegs2<-envfit(vares_cca2_veg ~., data=vegt2)$vectors[["pvals"]]%>% as.data.frame() %>% dplyr::select("Single Micop"=".")

vares_cca3_veg <- cca(spp_hell3 ~.,data=vegt3)
vegs3<-envfit(vares_cca3_veg ~., data=vegt3)$vectors[["pvals"]]%>% as.data.frame() %>% dplyr::select("Paired Micop"=".")

vares_cca4_veg <- cca(spp_hell4 ~.,data=vegt4)
vegs4<-envfit(vares_cca4_veg ~., data=vegt4)$vectors[["pvals"]]%>% as.data.frame() %>% dplyr::select("Kraken2"=".")

vegs<- cbind(vegs1, vegs2, vegs3, vegs4)
rownames(vegs)

```
- Plot vegetaion ccas and caps 
```{r, warning=FALSE, message=FALSE}
png("cap_veg.png",width=6,height=4,units="in",res=1200)
library(viridis)
par(mfrow=c(2,2),mar=c(2, 2, 1,1))

ordiplot(cap.veg1,type="n", main="QIIME2", cex.main=1)
text(cap.veg1, dis="cn", scaling="sites", cex=0.6, col="red")
#points(cap.veg1,display="sites",cex=0.5,pc=19,  col = pal2)
ordiellipse(cap.veg1, groups =meta1$Sites, col =pal, display = "sites",
            label=T, draw = "polygon", alpha = 50, label.cex=0.5, cex=0.7)
#text(x = vars_choosing1$CAP1,y = vars_choosing1$CAP2, 
#    cex=0.8, col="blue", labels=vars_choosing1$tax, font=3)
#text(cap.veg1, dis="species", scaling="sites", cex=0.5, col="black")


ordiplot(cap.veg2,type="n", main="SINGLE MICOP", cex.main=1)
text(cap.veg2, dis="cn", scaling="sites", cex=0.6, col="red")
#points(cap.veg2,display="sites",cex=0.5,pc=19,  col = pal2)
ordiellipse(cap.veg2, groups =meta2$Sites, col =pal, display = "sites",
            label=T, draw = "polygon", alpha = 50, label.cex=0.5, cex=0.7)
#text(x = vars_choosing2$CAP1,y = vars_choosing2$CAP2, 
#    cex=0.8, col="blue", labels=vars_choosing2$tax, font=3)

ordiplot(cap.veg3,type="n", main="PAIRED MICOP", cex.main=1)
text(cap.veg3, dis="cn", scaling="sites", cex=0.6, col="red")
#points(cap.veg3,display="sites",cex=0.5,pc=19,  col = pal2)
ordiellipse(cap.veg3, groups =meta3$Sites, col =pal, display = "sites",
            label=T, draw = "polygon", alpha = 50, label.cex=0.5, cex=0.7)

#text(x = vars_choosing3$CAP1,y = vars_choosing3$CAP2, 
#    cex=0.8, col="blue", labels=vars_choosing3$tax, font=3)

ordiplot(cap.veg4,type="n", main="KRAKEN2", cex.main=1)
text(cap.veg4, dis="cn", scaling="sites", cex=0.6, col="red")
#points(cap.veg4,display="sites",cex=0.5,pc=19,  col = pal2)
ordiellipse(cap.veg4, groups =meta4$Sites, col =pal, display = "sites",
            label=T, draw = "polygon", alpha = 50, label.cex=0.5, cex=0.7)
#text(x = vars_choosing4$CAP1,y = vars_choosing4$CAP2, 
#    cex=0.8, col="blue", labels=vars_choosing4$tax, font=3)



dev.off()


png("cap_veg_c.png",width=6,height=4,units="in",res=1200)
                     #plots
par(mfrow=c(2,2),mar=c(2, 2, 1, 1))

ordiplot(cap.veg1_c,type="n", main="QIIME2", cex.main=1)
text(cap.veg1_c, dis="cn", scaling="sites", cex=0.6, col="red")
#points(cap.veg1_c,display="sites",cex=0.5,pc=19,  col = pal2)
ordiellipse(cap.veg1_c, groups =meta1$Sites, col =pal, display = "sites",
            label=T, draw = "polygon", alpha = 50, label.cex=0.5, cex=0.7)
#text(x = vars_choosing1$CAP1,y = vars_choosing1$CAP2, 
#    cex=0.8, col="blue", labels=vars_choosing1$tax, font=3)
#text(cap.veg1_c, dis="species", scaling="sites", cex=0.5, col="black")


ordiplot(cap.veg2_c,type="n",main="SINGLE MICOP", cex.main=1)
text(cap.veg2_c, dis="cn", scaling="sites", cex=0.6, col="red")
#points(cap.veg2_c,display="sites",cex=0.5,pc=19,  col = pal2)
ordiellipse(cap.veg2_c, groups =meta2$Sites, col =pal, display = "sites",
            label=T, draw = "polygon", alpha = 50, label.cex=0.5, cex=0.7)
#text(x = vars_choosing2$CAP1,y = vars_choosing2$CAP2, 
#    cex=0.8, col="blue", labels=vars_choosing2$tax, font=3)

ordiplot(cap.veg3_c,type="n",main="PAIRED MICOP", cex.main=1)
text(cap.veg3_c, dis="cn", scaling="sites", cex=0.6, col="red")
#points(cap.veg3_c,display="sites",cex=0.5,pc=19,  col = pal2)
ordiellipse(cap.veg3_c, groups =meta3$Sites, col =pal, display = "sites",
            label=T, draw = "polygon", alpha = 50, label.cex=0.5, cex=0.7)

#text(x = vars_choosing3$CAP1,y = vars_choosing3$CAP2, 
#    cex=0.8, col="blue", labels=vars_choosing3$tax, font=3)

ordiplot(cap.veg4_c,type="n",main="KRAKEN2", cex.main=1)
text(cap.veg4_c, dis="cn", scaling="sites", cex=0.6, col="red")
#points(cap.veg4_c,display="sites",cex=0.5,pc=19,  col = pal2)
ordiellipse(cap.veg4_c, groups =meta4$Sites, col =pal, display = "sites",
            label=T, draw = "polygon", alpha = 50, label.cex=0.5, cex=0.7)
#text(x = vars_choosing4$CAP1,y = vars_choosing4$CAP2, 
#    cex=0.8, col="blue", labels=vars_choosing4$tax, font=3)



dev.off()


png("cca_veg.png",width=6,height=4,units="in",res=1200)
par(mfrow=c(2,2),mar=c(2, 2, 1, 1))

ordiplot(vares_cca1_veg,type="n", main="QIIME2", cex.main=1)
text(vares_cca1_veg, dis="cn", scaling="sites", cex=0.6, col="red")
#points(vares_cca1_veg,display="sites",cex=0.5,pc=19,  col = pal2)
ordiellipse(vares_cca1_veg, groups =meta1$Sites, col =pal, display = "sites",
            label=T, draw = "polygon", alpha = 50, label.cex=0.5, cex=0.7)
#text(x = vars_choosing1$CAP1,y = vars_choosing1$CAP2, 
#    cex=0.8, col="blue", labels=vars_choosing1$tax, font=3)
#text(vares_cca1_veg, dis="species", scaling="sites", cex=0.5, col="black")


ordiplot(vares_cca2_veg,type="n",main="SINGLE MICOP", cex.main=1)
text(vares_cca2_veg, dis="cn", scaling="sites", cex=0.6, col="red")
#points(vares_cca2_veg,display="sites",cex=0.5,pc=19,  col = pal2)
ordiellipse(vares_cca2_veg, groups =meta2$Sites, col =pal, display = "sites",
            label=T, draw = "polygon", alpha = 50, label.cex=0.5, cex=0.7)
#text(x = vars_choosing2$CAP1,y = vars_choosing2$CAP2, 
#    cex=0.8, col="blue", labels=vars_choosing2$tax, font=3)

ordiplot(vares_cca3_veg,type="n",main="PAIRED MICOP", cex.main=1)
text(vares_cca3_veg, dis="cn", scaling="sites", cex=0.6, col="red")
#points(vares_cca3_veg,display="sites",cex=0.5,pc=19,  col = pal2)
ordiellipse(vares_cca3_veg, groups =meta3$Sites, col =pal, display = "sites",
            label=T, draw = "polygon", alpha = 50, label.cex=0.5, cex=0.7)

#text(x = vars_choosing3$CAP1,y = vars_choosing3$CAP2, 
#    cex=0.8, col="blue", labels=vars_choosing3$tax, font=3)

ordiplot(vares_cca4,type="n",main="KRAKEN2", cex.main=1)
text(vares_cca4, dis="cn", scaling="sites", cex=0.6, col="red")
#points(vares_cca4_veg,display="sites",cex=0.5,pc=19,  col = pal2)
ordiellipse(vares_cca4, groups =meta4$Sites, col =pal, display = "sites",
            label=T, draw = "polygon", alpha = 50, label.cex=0.5, cex=0.7)
#text(x = vars_choosing4$CAP1,y = vars_choosing4$CAP2, 
#    cex=0.8, col="blue", labels=vars_choosing4$tax, font=3)



dev.off()

```


