---
title: "Figure 5 and S2"
author:
- name: Stephanie Hereira
  affiliation: Universidad Aut√≥noma de Tlaxcala
  email: sehereirap@uatx.mx
date: "`r format(Sys.time(), '%d - %m - %Y')`"
output:
  html_document:
    theme: flatly
    highlight: kate
    toc: yes
    toc_depth: 2
    toc_float: yes
  pdf_document:
    toc: yes
    toc_depth: '2'
---

## Figure 5 and S2: PCoAs

- Loading libraries, functions and data
```{r, warning=FALSE, message=FALSE}
library(tidyverse)
library(RColorBrewer)
library(cowplot)
library(broom)
library(geosphere)
library(reshape2)
library(readxl)
library(qiime2R)
source("../Code/functions_betadiv.R")
library(hilldiv)
library(hillR)


metadata<-read_excel("../Data/Metadatos.xlsx") %>% mutate_if(is.numeric, as.factor)
map<-metadata
table_single_micop<- read.delim("../Data/table_micop_single.txt") 
table_paired_micop<- read.delim("../Data/table_micop_paired.txt")
table_qiime2<- data.frame(read_qza("../Data/clustered_table_filter.qza")$data, 
                          check.names = F) %>% t() %>% as.data.frame(
                                ) %>% rownames_to_column(
                            var = "SampleID") %>% separate(
                              SampleID, c(
                                "id_metagenome", "R", "unmap", "Paired"), 
                              sep = "_")%>% inner_join(
                                metadata) %>% dplyr::select(
                                -id_metagenome:-Paired, 
                                -id_sequence:-id_fisicoq, -Sites, 
                                -Names) %>% column_to_rownames(
                                  var="SampleID") %>% t(
                                  ) %>% as.data.frame() %>% mutate_all(as.numeric)

table_fungi<- read.delim("../Data/table_kraken.txt", 
                        row.names = 1, check.names = F) %>% dplyr::select_if(
                           is.numeric)%>% t() %>% as.data.frame() %>% rownames_to_column(
                             var = "id_sequence") %>% separate(
                               ., "id_sequence", c(
                                 "kraken", "fungi", "id_metagenome", "report","bracken"),
                               sep = "_") %>% dplyr::select(
                                 -kraken, -fungi, -report, -bracken) %>% full_join(
                                 metadata) %>% dplyr::select(
                                   -id_sequence:-Transecto, -id_metagenome,
                                   -Sites, -id_new, -Names,  -id_fisicoq) %>% column_to_rownames(var = "SampleID") %>% t() %>% as.data.frame() %>% mutate_all(as.numeric)
```

- PCoA calculation on list of tables

```{r, warning=FALSE, message=FALSE}
#make a list of tables
otu <- list(table_qiime2, table_single_micop, table_paired_micop,  table_fungi)
# matching to map to set same order
otu_match<- lapply(otu, otu.match) 
#remove singletons
otu_single<- lapply(otu_match, otu.single) 
#Calculate Horn dissimilaties
dist<- lapply(otu_single, beta_div_dist_hill, q=1)
#Normalize for Bray-curtis distance
otu_norm<- lapply(otu_single, otu.norm)
#Calculate Bray-Curtis dissimilarities
dist2<- lapply(otu_norm, beta_div_dist_bray)
# PCoA Horn
pcoa<- lapply(dist, pcoa_all) 
pcoa.axes <- lapply(dist, pcoa_axes) #obtain pcoa axes
pcoa.eigval <- lapply(dist, pcoa_eigval) #obtain pcoa eigvalues
# PCoA Bray-curtis
pcoa2 <-lapply(dist2, pcoa_all) 
pcoa.axes2 <- lapply(dist2, pcoa_axes) #obtain pcoa axes
pcoa.eigval2 <- lapply(dist2, pcoa_eigval) #obtain pcoa eigvalues
```

- perMANOVA and betadisper calculations
```{r, warning=FALSE, message=FALSE}
#permanova's
set.seed(124)
perm_qiime2<- permanova_beta(dist[[1]], metadata = metadata)
perm_micop_single<- permanova_beta(dist[[2]], metadata = metadata)
perm_micop_paired<- permanova_beta(dist[[3]], metadata = metadata)
perm_fungi<- permanova_beta(dist[[4]], metadata = metadata)

perm_qiime22<- permanova_beta(dist2[[1]], metadata = metadata)
perm_micop_single2<- permanova_beta(dist2[[2]], metadata = metadata)
perm_micop_paired2<- permanova_beta(dist2[[3]], metadata = metadata)
perm_fungi2<- permanova_beta(dist2[[4]], metadata = metadata)

#betadisper
set.seed(125)
permd_qiime2<- permdisp_beta(dist[[1]], metadata = metadata)
permd_micop_single<- permdisp_beta(dist[[2]], metadata = metadata)
permd_micop_paired<- permdisp_beta(dist[[3]], metadata = metadata)
permd_fungi<- permdisp_beta(dist[[4]], metadata = metadata)

permd_qiime22<- permdisp_beta(dist2[[1]], metadata = metadata)
permd_micop_single2<- permdisp_beta(dist2[[2]], metadata = metadata)
permd_micop_paired2<- permdisp_beta(dist2[[3]], metadata = metadata)
permd_fungi2<- permdisp_beta(dist2[[4]], metadata = metadata)
```


- PCoA plots Horn and Bray-curtis
```{r, warning=FALSE, message=FALSE, fig.width=9, fig.height=7}
qiime2_pcoa <- pcoa_plot(pcoa[[1]])+
  #xlab(paste("PCo1 (", pcoa.eigval[[1]]$Eigval[1], "%)", sep = "")) +
  #ylab(paste("PCo2 (", pcoa.eigval[[1]]$Eigval[2], "%)", sep = "")) +
  ggtitle("GENEIOS+UNITE")+  
 # xlab("DIM1")+
  #ylab("DIM2")+
  labs(title = paste("adonis: F = ", signif(perm_qiime2$F[1], 3), ",",
                     "p-value = ",signif(perm_qiime2$`Pr(>F)`[1], 5), 
                     "\n betadisper: F = ", signif(permd_qiime2$tab$F[1], 2), ",",
                     "p-value = ",signif(permd_qiime2$tab$`Pr(>F)`[1], 5)))+theme(
                       legend.text = element_text(size = 20), legend.title = element_text(size = 20),
                       axis.title = element_text(size = 16),
                       plot.title = element_text(hjust = 1, size = 12),
                       legend.key.size = unit(1, "cm"))+ guides(
                         fill = guide_legend(override.aes = list(size = 8)))+
   geom_vline(xintercept = 0, linetype = 2, color="#88929b") +   #lines-cross
  geom_hline(yintercept = 0, linetype = 2, color="#88929b") 

qiime2_pcoa2<- pcoa_plot(pcoa2[[1]])+
  #xlab(paste("PCo1 (", pcoa.eigval[[1]]$Eigval[1], "%)", sep = "")) +
  #ylab(paste("PCo2 (", pcoa.eigval[[1]]$Eigval[2], "%)", sep = "")) +
  ggtitle("GENEIOUS+UNITE")+  
  #xlab("DIM1")+
  #ylab("DIM2")+
  labs(title = paste("adonis: F = ", signif(perm_qiime22$F[1], 3), ",",
                     "p-value = ",signif(perm_qiime22$`Pr(>F)`[1], 5), 
                     "\n betadisper: F = ", signif(permd_qiime22$tab$F[1], 2), ",",
                     "p-value = ",signif(permd_qiime22$tab$`Pr(>F)`[1], 5)))+theme(
                       legend.text = element_text(size = 20), legend.title = element_text(size = 20),
                       axis.title = element_text(size = 16),
                       plot.title = element_text(hjust = 1, size = 12),
                       legend.key.size = unit(1, "cm"))+ guides(
                         fill = guide_legend(override.aes = list(size = 8)))+
   geom_vline(xintercept = 0, linetype = 2, color="#88929b") +   #lines-cross
  geom_hline(yintercept = 0, linetype = 2, color="#88929b") 


single_pcoa <- pcoa_plot(pcoa[[2]])+
  xlab(paste("DIM1 ", pcoa.eigval[[2]]$Eigval[1], "%", sep = "")) +
  ylab(paste("DIM2 ", pcoa.eigval[[2]]$Eigval[2], "%", sep = "")) +
  ggtitle("SINGLE MICOP")+
  labs(title = paste("adonis: F = ", signif(perm_micop_single$F[1], 3), ",",
                     "p-value = ",signif(perm_micop_single$`Pr(>F)`[1], 5), 
                     "\n betadisper: F = ", signif(permd_micop_single$tab$F[1], 2), ",",
                     "p-value = ",signif(permd_micop_single$tab$`Pr(>F)`[1], 5)))+theme(
                       legend.text = element_text(size = 20), legend.title = element_text(size = 20),
                       axis.title = element_text(size = 16),
                       plot.title = element_text(hjust = 1, size = 12),
                       legend.key.size = unit(2, "cm"))+ guides(
                         fill = guide_legend(override.aes = list(size = 8)))+
   geom_vline(xintercept = 0, linetype = 2, color="#88929b") +   #lines-cross
  geom_hline(yintercept = 0, linetype = 2, color="#88929b") 

single_pcoa2 <- pcoa_plot(pcoa2[[2]])+
   xlab(paste("DIM1 ", pcoa.eigval2[[2]]$Eigval[1], "%", sep = "")) +
   ylab(paste("DIM2 ", pcoa.eigval2[[2]]$Eigval[2], "%", sep = "")) +
  ggtitle("SINGLE MICOP")+
  labs(title = paste("adonis: F = ", signif(perm_micop_single2$F[1], 3), ",",
                     "p-value = ",signif(perm_micop_single2$`Pr(>F)`[1], 5), 
                     "\n betadisper: F = ", signif(permd_micop_single2$tab$F[1], 2), ",",
                     "p-value = ",signif(permd_micop_single2$tab$`Pr(>F)`[1], 5)))+theme(
                       legend.text = element_text(size = 20), legend.title = element_text(size = 20),
                       axis.title = element_text(size = 16),
                       plot.title = element_text(hjust = 1, size = 12),
                       legend.key.size = unit(2, "cm"))+ guides(
                         fill = guide_legend(override.aes = list(size = 8)))+
   geom_vline(xintercept = 0, linetype = 2, color="#88929b") +   #lines-cross
  geom_hline(yintercept = 0, linetype = 2, color="#88929b") 

paired_pcoa <- pcoa_plot(pcoa[[3]])+
  xlab(paste("DIM1 ", pcoa.eigval[[3]]$Eigval[2], "%", sep = "")) +
  ylab(paste("DIM2 ", pcoa.eigval[[3]]$Eigval[3], "%", sep = "")) +
#  xlab("DIM1")+
 # ylab("DIM2")+
  ggtitle("PAIRED MICOP")+
  labs(title = paste("adonis: F = ", signif(perm_micop_paired$F[1], 3), ",",
                     "p-value = ",signif(perm_micop_paired$`Pr(>F)`[1], 5), 
                     "\n betadisper: F = ", signif(permd_micop_paired$tab$F[1], 2), ",",
                     "p-value = ",signif(permd_micop_paired$tab$`Pr(>F)`[1], 5)))+theme(
                       legend.text = element_text(size = 20), legend.title = element_text(size = 20),
                       axis.title = element_text(size = 16),
                       plot.title = element_text(hjust = 1, size = 12),
                       legend.key.size = unit(2, "cm"))+ guides(
                         fill = guide_legend(override.aes = list(size = 8)))+
   geom_vline(xintercept = 0, linetype = 2, color="#88929b") +   #lines-cross
  geom_hline(yintercept = 0, linetype = 2, color="#88929b") 

  
paired_pcoa2 <- pcoa_plot(pcoa2[[3]])+
  xlab(paste("DIM1 ", pcoa.eigval2[[3]]$Eigval[1], "%", sep = "")) +
    ylab(paste("DIM2 ", pcoa.eigval2[[3]]$Eigval[2], "%", sep = "")) +
 # xlab("DIM1")+
  #ylab("DIM2")+
  ggtitle("PAIRED MICOP")+
  labs(title = paste("adonis: F = ", signif(perm_micop_paired2$F[1], 3), ",",
                     "p-value = ",signif(perm_micop_paired2$`Pr(>F)`[1], 5), 
                     "\n betadisper: F = ", signif(permd_micop_paired2$tab$F[1], 2), ",",
                     "p-value = ",signif(permd_micop_paired2$tab$`Pr(>F)`[1], 5)))+theme(
                       legend.text = element_text(size = 20), legend.title = element_text(size = 20),
                       axis.title = element_text(size = 16),
                       plot.title = element_text(hjust = 1, size = 12),
                       legend.key.size = unit(2, "cm"))+ guides(
                         fill = guide_legend(override.aes = list(size = 8)))+
   geom_vline(xintercept = 0, linetype = 2, color="#88929b") +   #lines-cross
  geom_hline(yintercept = 0, linetype = 2, color="#88929b") 




kraken_pcoa <- pcoa_plot(pcoa[[4]])+
#  scale_x_continuous(limits = c(-0.01,0.01))+
 # scale_y_continuous(limits = c(-0.01,0.01))+
  xlab(paste("DIM1 ", pcoa.eigval[[4]]$Eigval[2], "%", sep = "")) +
  ylab(paste("DIM2 ", pcoa.eigval[[4]]$Eigval[3], "%", sep = "")) +
  ggtitle("KRAKEN2")+
  labs(title = paste("adonis: F = ", signif(perm_fungi$F[1], 3), ",",
                     "p-value = ",signif(perm_fungi$`Pr(>F)`[1], 5), 
                     "\n betadisper: F = ", signif(permd_fungi$tab$F[1], 2), ",",
                     "p-value = ",signif(permd_fungi$tab$`Pr(>F)`[1], 5)))+theme(
                       legend.text = element_text(size = 20), legend.title = element_text(size = 20),
                       axis.title = element_text(size = 16),
                       plot.title = element_text(hjust = 1, size = 12),
                       legend.key.size = unit(2, "cm"))+ guides(
                         fill = guide_legend(override.aes = list(size = 8)))+
   geom_vline(xintercept = 0, linetype = 2, color="#88929b") +   #lines-cross
  geom_hline(yintercept = 0, linetype = 2, color="#88929b") 

kraken_pcoa2 <- pcoa_plot(pcoa2[[4]])+
  #  scale_x_continuous(limits = c(-0.01,0.01))+
  # scale_y_continuous(limits = c(-0.01,0.01))+
   xlab(paste("DIM1 ", pcoa.eigval2[[4]]$Eigval[2], "%", sep = "")) +
    ylab(paste("DIM2 ", pcoa.eigval2[[4]]$Eigval[3], "%", sep = "")) +
  ggtitle("KRAKEN2")+
  labs(title = paste("adonis: F = ", signif(perm_fungi2$F[1], 3), ",",
                     "p-value = ",signif(perm_fungi2$`Pr(>F)`[1], 5), 
                     "\n betadisper: F = ", signif(permd_fungi2$tab$F[1], 2), ",",
                     "p-value = ",signif(permd_fungi2$tab$`Pr(>F)`[1], 5)))+theme(
                       legend.text = element_text(size = 20), legend.title = element_text(size = 20),
                       axis.title = element_text(size = 16),
                       plot.title = element_text(hjust = 1, size = 12),
                       legend.key.size = unit(2, "cm"))+ guides(
                         fill = guide_legend(override.aes = list(size = 8)))+
   geom_vline(xintercept = 0, linetype = 2, color="#88929b") +   #lines-cross
  geom_hline(yintercept = 0, linetype = 2, color="#88929b") 
    

leg<- get_legend(qiime2_pcoa)
leg2<- get_legend(qiime2_pcoa2)


first<-plot_grid(qiime2_pcoa+theme(legend.position = "none")+theme(aspect.ratio =6/10),  
                 single_pcoa+theme(legend.position = "none")+theme(aspect.ratio =6/10), 
                 paired_pcoa+theme(legend.position = "none")+theme(aspect.ratio =6/10),   
                 kraken_pcoa+theme(legend.position = "none")+theme(aspect.ratio =6/10), 
                 ncol = 2, nrow = 2, rel_widths = c(1,1,1,1),
                 align = "v",
                 labels = c("A) GENEIOUS+UNITE", "B) SINGLE MICOP",
                            "C) PAIRED MICOP", "D) KRAKEN2"),hjust = 0)
second<-plot_grid(qiime2_pcoa2+theme(legend.position = "none")+theme(aspect.ratio =6/10),  
                 single_pcoa2+theme(legend.position = "none")+theme(aspect.ratio =6/10),
                 paired_pcoa2+theme(legend.position = "none")+theme(aspect.ratio =6/10),   
                 kraken_pcoa2+theme(legend.position = "none")+theme(aspect.ratio =6/10), 
                 ncol = 2, nrow = 2, rel_widths = c(1,1,1,1),
                 align = "v",
                 labels = c("A) GENEIOUS+UNITE", "B) SINGLE MICOP",
                            "C) PAIRED MICOP", "D) KRAKEN2"),hjust = 0)

pcoas_plot <- plot_grid(first,leg,ncol = 2, rel_widths = c(.9,.1), align = "hv")
pcoas_plot
pcoas_plot2 <- plot_grid(second,leg,ncol = 2, rel_widths = c(.9,.1), align = "hv")
pcoas_plot2
ggsave("Figure5_pcoas_plot_horn_final.pdf",width =11, height =7, dpi = 600, plot = pcoas_plot, device = "pdf")
ggsave("FigureS2_pcoas_plot_bray.png",width =11, height =7, dpi = 600, plot = pcoas_plot2, device = "png")
```

#partial mantel tests

```{r}
#environmental data
metadata_secas<- read_excel("../Data/Metadatos.xlsx", sheet = "secas-marzo")
fq_secas<- read_excel("../Data/fisicoq.xlsx", sheet = "seca")
fq_secas2<- read.csv("../Data/fisicoq-la.csv")

meta_fq_secas<- metadata_secas %>% full_join(fq_secas) #%>%  select(Sites:id_new,pH, MO, N, P) %>% mutate(Season="Dry")
meta_fq_secas_all<- metadata_secas %>% full_join(fq_secas) %>% full_join(fq_secas2, by = "SampleID")

env.16S=meta_fq_secas_all
df<- env.16S%>% dplyr::select(SampleID, pH:Mn, moisture, WHC:CONDUC, ARCILLA:ARENA) 
dfs=data.frame(df[1],scale(df[,2:17], center = T, scale = T)) %>% dplyr::select(
  SampleID,P,K,Ca,Mg,moisture,WHC, LIMO) %>% 
 column_to_rownames(var = "SampleID")

envs_dist<- dist(dfs, method = "euclidean")

#vegetation data

veg=read.delim("../Data/vegeta.tsv")

veg_dist<- dist(veg, method = "euclidean")

library(geosphere)
coords<- read_csv("../Data//coord.csv")

coords_mat<- coords%>% mutate(P=paste0("P",pol),
                              S=paste0("S", Sitio),
                              T=paste0("T", Transecto)
) %>% unite(
  "SampleID", 
  P:T, sep = "") %>% select(
    SampleID, Longitude, Latitude) %>% column_to_rownames(var = "SampleID") %>% as.matrix() 

distance<- distm(coords_mat)/1000
colnames(distance)<- rownames(coords_mat)
rownames(distance)<- rownames(coords_mat)
distance_complete<- distance

```

```{r}
# partial mantel tests horn
set.seed(124)
mantel.partial(dist[[1]], envs_dist, distance_complete, method = "spearman")
mantel.partial(dist[[1]], veg_dist, distance_complete, method = "spearman")

mantel.partial(dist[[2]], envs_dist, distance_complete, method = "spearman")
mantel.partial(dist[[2]], veg_dist, distance_complete, method = "spearman")

mantel.partial(dist[[3]], envs_dist, distance_complete, method = "spearman")
mantel.partial(dist[[3]], veg_dist, distance_complete, method = "spearman")

mantel.partial(dist[[4]], envs_dist, distance_complete, method = "spearman")
mantel.partial(dist[[4]], veg_dist, distance_complete, method = "spearman")
```

```{r}
# partial mantel tests bray
mantel.partial(dist2[[1]], envs_dist, distance_complete, method = "spearman")
mantel.partial(dist2[[1]], veg_dist, distance_complete, method = "spearman")

mantel.partial(dist2[[2]], envs_dist, distance_complete, method = "spearman")
mantel.partial(dist2[[2]], veg_dist, distance_complete, method = "spearman")

mantel.partial(dist2[[3]], envs_dist, distance_complete, method = "spearman")
mantel.partial(dist2[[3]], veg_dist, distance_complete, method = "spearman")

mantel.partial(dist2[[4]], envs_dist, distance_complete, method = "spearman")
mantel.partial(dist2[[4]], veg_dist, distance_complete, method = "spearman")
```

# hierarchical partitioning

```{r}

```

